### ISSUE TITLE
Electrical Module: A NetworkXNoPath occurred

### ISSUE THREAD

**Issue by dthunder746** [DTOcean/dtocean#1]
Hey,

Using the V 1.0 .exe from the SETIS site, and the Wave Device Example data from the database.

Performing a simulation with the Hydro and Electrical modules, and the Economic assessment I get the following error when beginning the Electrical module:

```
2017-01-20 16:13:23,404 - INFO - dtocean_electrical.main - Begin main run...
2017-01-20 16:13:23,404 - INFO - dtocean_electrical.grid.grid_processing - Merging bathymetry...
2017-01-20 16:13:23,513 - INFO - dtocean_electrical.grid.grid_processing - Creating bathymetry grid points...
2017-01-20 16:13:38,200 - INFO - dtocean_electrical.grid.grid_processing - Checking distance to neighbours...
2017-01-20 16:13:49,427 - INFO - dtocean_electrical.grid.grid_processing - Checking gradient to neighbours...
2017-01-20 16:13:52,714 - INFO - dtocean_electrical.grid.grid_processing - Building network graph...
2017-01-20 16:13:53,851 - INFO - dtocean_electrical.grid.grid_processing - Checking for exclusion zones...
2017-01-20 16:13:54,782 - INFO - dtocean_electrical.grid.grid_processing - Number of points removed in exclusion zones: 20
2017-01-20 16:13:54,783 - INFO - dtocean_electrical.grid.grid_processing - Checking gradient constraints...
2017-01-20 16:13:55,108 - INFO - dtocean_electrical.grid.grid_processing - Checking equipment soil compatibility...
2017-01-20 16:13:56,619 - INFO - dtocean_electrical.grid.grid_processing - Constraints checked...
2017-01-20 16:13:56,806 - INFO - dtocean_electrical.grid.grid_processing - Grid prepared...
2017-01-20 16:13:57,500 - INFO - dtocean_electrical.main - Building network...
2017-01-20 16:13:58,065 - CRITICAL - dtocean_gui.main - A NetworkXNoPath occurred: node 16056 not reachable from 23272
2017-01-20 16:13:58,220 - CRITICAL - dtocean_gui.main -   File "C:\DTOcean\lib\site-packages\dtocean_gui\main.py", line 185, in run
    self._project)
  File "C:\DTOcean\lib\site-packages\dtocean_core\menu.py", line 457, in execute_current
    allow_unavailable=allow_unavailable)
  File "C:\DTOcean\lib\site-packages\dtocean_core\menu.py", line 168, in _execute
    allow_unavailable=allow_unavailable)
  File "C:\DTOcean\lib\site-packages\dtocean_core\core.py", line 1769, in execute_interface
    interface = core.connect_interface(project, interface)
  File "C:\DTOcean\lib\site-packages\dtocean_gui\core.py", line 267, in connect_interface
    interface = super(GUICore, self).connect_interface(project, interface)
  File "C:\DTOcean\lib\site-packages\dtocean_core\core.py", line 1244, in connect_interface
    interface.connect()
  File "C:\DTOcean\lib\site-packages\dtocean_core\interfaces\electrical.py", line 759, in connect
    solution, installation_tool = elec.run_module(plot=False)
  File "C:\DTOcean\lib\site-packages\dtocean_electrical\main.py", line 156, in run_module
    electrical_design.set_design_limits()
  File "C:\DTOcean\lib\site-packages\dtocean_electrical\optim_codes\optimiser.py", line 339, in set_design_limits
    distance_matrix = self.calculate_device_spacings(local_cp)
  File "C:\DTOcean\lib\site-packages\dtocean_electrical\optim_codes\optimiser.py", line 497, in calculate_device_spacings
    self.meta_data.grid.graph  # this is unfiltered graph
  File "C:\DTOcean\lib\site-packages\dtocean_electrical\optim_codes\array_layout.py", line 340, in calculate_distance_dijkstra
    in itertools.product(np.asarray(layoutlist),repeat=2)])
  File "C:\DTOcean\lib\site-packages\networkx\algorithms\shortest_paths\weighted.py", line 133, in dijkstra_path_length
    "node %s not reachable from %s" % (source, target))
```

Any ideas?

**Comment by H0R5E** [comment]
Hi, did you follow the instructions in the README that came with the DB? Specifically it says this about the wave example with those two modules:

> ********************
> Wave Device Example
> ********************
> 
> The hydrodynamic model corresponding to the example wave device can be found in the file wec_solution.h5.  This file should be copied to a new directory on the local machine.  The directory path can either be entered via the DTOcean GUI or into the DTOcean database, in column beta.device(wave_data_directory).
> 
> Within the DTOcean GUI, the following options should be set:
> 
> Hydrodynamics
> -------------
> Lease Area Boundary Padding = 400
> Minimum Q-Factor = 0.9
> Power Histogram Bin Width = 0.01
> User Option for Array Layout = Staggered
> 
> Electrical Sub Systems
> ----------------------
> Umbilical Cable Safety Factor = 0.0
> Network Configuration = Radial
> Maximum Seabed Gradient = 14.00

My best guess is that your maximum seabed gradient is set too low.

Mat

**Comment by dthunder746** [comment]
Yep, followed the text file. I tried increasing the seabed gradient from 14, but even going up to 1000 results in the exact same error.

I noticed that there is a big change in depth on the boundary between the lease area and cable corridor, so I changed the bathymetry data for the lease area and cable corridor to be a constant -70m. This fixed the issue, so thanks for the tip about the seabed gradient.
